test/system/src/test/java/io/pravega/test/system/ControllerFailoverTest.java

```diff
@@ -46,14 +48,15 @@
 @Slf4j
 @RunWith(SystemTestRunner.class)
 public class ControllerFailoverTest {
-    private static final String TEST_CONTROLLER_SERVICE_NAME = "testcontroller";
     private static final ScheduledExecutorService EXECUTOR_SERVICE = Executors.newScheduledThreadPool(5);
+    private Service controllerService1 = null;
+    private URI controllerURIDirect = null;
 
     @Environment
-    public static void setup() throws InterruptedException, MarathonException, URISyntaxException {
+    public static void initialize() throws InterruptedException, MarathonException, URISyntaxException, ExecutionException {
```

Comment:
Why is it changing from setup to initialize? The change seems to be gratuitous.

Reply:
This is changed because another method  with name `setup`  was added which ensures everything is deployed and collects the `controllerUri` to use it in @Test method. The  instances instantiated in a method with `@Environment` cannot be accessed else where in code. So, this leads to defining new method, which just collects the required information of the services deployed and pass into on the code wherever required.

Reply:
It is odd to have both a setup method and an initialize method, they in principle imply very similar concepts, which is to do any necessary wiring and setting that the instance needs to start executing. To me this smells like a problem with the abstractions.

Reply:
renamed the `setup` to `getControllerInfo`  as we use it for the same.